templates/webpack.config.js

```diff
@@ -95,34 +114,51 @@ const generateConfig = () => {
       }
     case allServicesConfig:
       return {
+        // todo: apply env.js to all service entries
         entry: getAllServicesEntries(),
         output: {
           filename: `[name]`,
-          path: getServicesPath()
+          path: getServicesPath(),
+          environment: codeEngineEnvironment
         }
       }
     case serviceConfig:
       return {
-        entry: `${getServicePath(service, true)}/${service}`,
+        entry: {
+          index: [
+            path.resolve(__dirname, "polyfills/env.js"),
+            `${getServicePath(service, true)}/${service}`,
+          ],
+        },
+        plugins: [
+          new webpack.ProvidePlugin({
```

Comment:
do we not need to apply these plugins to `allServicesConfig` and libraries?

Reply:
Yea, I left a todo comment below